src/NetCore/RequestTools.cs

```diff
@@ -35,6 +35,10 @@ internal static bool Validate(HttpRequest req, ISigningAlgorithm algorithm, IApp
             {
                 string contentMd5 = h.Get(Headers.ContentMD5);
                 var builder = new CannonicalRepresentationBuilder();
+
+                var url = h.Get(Headers.XOriginalUrl);
+                url = string.IsNullOrWhiteSpace(url) ? req.GetEncodedUrl() : $"https://{req.Host.Host}{url}";
```

Comment:
`$"https://{req.Host.Host}{url}"` this is not correct - schema can be different, port can different and `url` is empty at the moment as a result request URI will be wrong

Reply:
Ok, `url` will not be empty, but other statements are valid